state/txindex/kv/kv.go

```diff
@@ -42,6 +48,240 @@ type TxIndex struct {
 	log log.Logger
 }
 
+func (txi *TxIndex) Prune(retainHeight int64) (int64, int64, error) {
```

Comment:
IMPORTANT: There is a problem in this version of `Prune`. `PruneOld` also deletes the events, whereas this version doesn't